jupyterhub_config.py

```diff
@@ -27,3 +27,22 @@
 c.KubeSpawner.user_storage_pvc_ensure = False
 
 c.JupyterHub.allow_named_servers = True
+
+c.KubeSpawner.profile_list = [
+    {
+        'display_name': 'Training Env - Python',
+        'default': True,
+        'kubespawner_override': {
+            'singleuser_image_spec': 'training/python:label',
+            'cpu_limit': 0.5,
+        },
+        'description': 'Something description of what is going on here, maybe a <a href="#">link too!</a>'
+    }, {
+        'display_name': 'Training Env - Datascience',
+        'kubespawner_override': {
+            'singleuser_image_spec': 'training/datascience:label',
+            'cpu_limit': 0.2,
+        },
+        'description': 'Something description of how this is different, maybe a <a href="#">link too!</a>'
+    }
```

Comment:
I'll go to sleep before I think this through properly, but I'm considering the implications of pre-pulling images in z2jh-k8s. I'll also need to figure out what it means to have a jupyterhub_config.py file within the kubespawner project.

Reply:
It might be good to move this config file to an `examples` directory with a README.

Reply:
I agree I am not really fan of exposing kubespawner in jupyterhub config, but that was the easiest and safest way of doing so. Or we would have had to expose all settings kubeswpaner propose, and having to maintain it manually at each evolution.
But since each entry is a dict, we can add a easier key that would only gives a subset of the settings, the most common (image, cpu/memory?)
But at the end, we can tell it is an advanced settings and read carefully the documentation.

The thing is, exposing kubesettings in jupyterhub conf means each api break (on kubespawner), for instance on major change, will impact this setting. That fine for me, but it is a dependency to take into account.

Reply:
This is how Spawners are intended to be configured, so it's the right way to do it, and the design goal of the configuration system used by Jupyter and JupyterHub. All settings tagged with `config=True` are automatically exposed in jupyterhub_config.py, without taking any additional steps.

Note that this file only serves as *documentation* of configurable options, and has no influence on actual behavior, which is why I suggested moving it to examples.

Reply:
oh, maybe we want to restrict `kubespawner_override` with the option that has a `config=True` ? If the current behavior is ok for you, I'm fine too !